engine/apps/public_api/serializers/routes.py

```diff
@@ -2,14 +2,125 @@
 
 from apps.alerts.models import AlertReceiveChannel, ChannelFilter, EscalationChain
+from apps.base.messaging import get_messaging_backend_from_id, get_messaging_backends
 from common.api_helpers.custom_fields import OrganizationFilteredPrimaryKeyRelatedField
 from common.api_helpers.exceptions import BadRequest
 from common.api_helpers.mixins import OrderedModelSerializerMixin
 
 
-class ChannelFilterSerializer(OrderedModelSerializerMixin, serializers.ModelSerializer):
+class BaseChannelFilterSerializer(OrderedModelSerializerMixin, serializers.ModelSerializer):
+    """Base Channel Filter serializer with validation methods"""
+
+    def __init__(self, *args, **kwargs):
+        """Update existing fields of the serializer with messaging backends fields"""
+
+        super().__init__(*args, **kwargs)
+        for backend_id, _ in get_messaging_backends():
+            backend = get_messaging_backend_from_id(backend_id)
+            if backend is None:
+                continue
+            field = backend_id.lower()
+            self._declared_fields[field] = serializers.DictField(required=False)
+            self.Meta.fields.append(field)
+
+    def to_representation(self, instance):
+        result = super().to_representation(instance)
+        result["slack"] = {"channel_id": instance.slack_channel_id}
+        result["telegram"] = {"id": instance.telegram_channel.public_primary_key if instance.telegram_channel else None}
+        # add representation for other messaging backends
+        for backend_id, _ in get_messaging_backends():
+            backend = get_messaging_backend_from_id(backend_id)
+            if backend is None:
+                continue
+            field = backend_id.lower()
+            channel_id = None
+            if instance.notification_backends and instance.notification_backends.get(backend_id, {}).get("channel"):
+                channel_id = instance.notification_backends[backend_id]["channel"]
+            result[field] = {"id": channel_id}
+        return result
+
+    def _correct_validated_data(self, validated_data):
+        organization = self.context["request"].auth.organization
+
+        slack_field = validated_data.pop("slack", {})
+        if "channel_id" in slack_field:
+            validated_data["slack_channel_id"] = self._validate_slack_channel_id(slack_field.get("channel_id"))
+
+        telegram_field = validated_data.pop("telegram", {})
+        if "id" in telegram_field:
+            validated_data["telegram_channel"] = self._validate_telegram_channel(telegram_field.get("id"))
+
+        notification_backends = {}
+        for backend_id, _ in get_messaging_backends():
+            field = backend_id.lower()
+            backend_field = validated_data.pop(field, {})
+            if backend_field:
+                backend = get_messaging_backend_from_id(backend_id)
+                if backend is None:
+                    continue
+                channel_id = backend_field.get("id")
```

Comment:
I was thinking something like `"MSTEAMS": {"channel": "public-id"}`, that would match what we expect in the backend and then no extra transformation would be needed (so we pass it through, being the backend implementation being responsible of validating the fields/data).
Although on the other hand, since we use `channel_id` for slack, why not use `channel_id` for telegram and other backends too, for consistency? Thoughts?

Reply:
For slack we use channel id, generated by slack, and it is not equal to our public pk. That's why these names for ids should be different - `channel_id` for external data, `id` - for our pks.